src/Event/DefaultRoleEvent.php

```diff
@@ -41,43 +59,46 @@ public function getRoles() {
   /**
    * {@inheritdoc}
    */
-  public function addRole($name, array $properties) {
-    if (array_key_exists($name, $this->roles)) {
-      throw new \InvalidArgumentException("The '$name' role already exists.");
-    }
-    $this->validate($name, $properties);
+  public function addRole(array $properties) {
+    $this->validate($properties);
 
-    // Provide default value for the role type.
-    if (empty($properties['role_type'])) {
-      $properties['role_type'] = OgRoleInterface::ROLE_TYPE_STANDARD;
+    if (array_key_exists($properties['name'], $this->roles)) {
```

Comment:
I think this is still a bit weird that `DefaultRoleEvent` is dealing with validation of OgRole creation. I think that maybe we can completely remove this method, and possibly move validation to OgRole .

See comment I'll add in OgEventSubscriber:: provideDefaultRoles ;)